### PR TITLE
Fix get_dtype and convert_into_dtypes (#769)

### PR DESCRIPTION
**Pull request by IT-Forrest**
# What does this PR do?

This is the cherry-pick of the Optimum-habana commit https://github.com/huggingface/optimum-habana/pull/769. The fix consists in extracting the dtype of the first logit tensor and set it as the target dtype. We used to extract the dtypes of all logit tensors, leading to nested tuples that led to this issue. There is no reason for logits to have different dtypes, so we can simplify the dtype extraction logic.

Fixes # (https://github.com/huggingface/optimum-habana/issues/754)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by libinta — APPROVED** [review]

**Comment by astachowiczhabana** [comment]
https://github.com/huggingface/optimum-habana/pull/769
